### PR TITLE
adding DegenHive adapter

### PR DESCRIPTION
**Pull request by alienHunterOnMars**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Logo (High resolution, will be shown with rounded borders):
Group 1000006745

##### Treasury Addresses (if the protocol has treasury)
SUI Treasury --> 0x5a79f26a90cf6a652cf45ad3dfbe1e3565503e70d866e3149292ccbc8fcd1420

##### Token address and ticker if any:
 DSUI (liquid staked SUI) : https://suivision.xyz/coin/0x50c2216a078d3bdf5081fe436df9f42dfdbe538ebd9c935913ce2436362cff90::dsui::DSUI
 HIVE (gov token) : https://suivision.xyz/coin/0x50c2216a078d3bdf5081fe436df9f42dfdbe538ebd9c935913ce2436362cff90::hive::HIVE
 HONEY (gov token with 3% burn tax) : https://suivision.xyz/coin/0x50c2216a078d3bdf5081fe436df9f42dfdbe538ebd9c935913ce2436362cff90::honey::HONEY

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/DegenHive: 
        
 
  
    
    
    Error: Cannot find module '@mysten/sui.js/transactions'
    Require stack:
    - /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/DegenHive/index.js
    - /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js
    Truncated error stack:
    
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/DegenHive/index.js:3:28)
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'

**Comment by llamatester** [comment]
Error while running adapter at projects/DegenHive: 
        
 
  
    
    
    SyntaxError: Cannot convert 75016650459.981931000000000 to a BigInt
    Truncated error stack:
        at convertToBigInt (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/generalUtil.js:57:16)
        at BigInt (<anonymous>)
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/DegenHive/index.js:62:9)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:173:9

**Comment by llamatester** [comment]
The adapter at projects/DegenHive exports TVL: 
        
 
 
    sui                       932
    
    total                    932.25111792965

**Comment by llamatester** [comment]
The adapter at projects/DegenHive exports TVL: 
        
 
 
    sui                       930
    
    total                    929.7068001406601

File: projects/DegenHive/index.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    pools {
+      meme_pool_addr
+    }
+  }
+}`
+
+ 
+
+async function tvl(api) {
+    const sui_pools = await graphQLClientMainnet.request(GET_SUI_POOLS, {})
```
Comment (g1nt0ki) [inline projects/DegenHive/index.js:27]: can you use `cachedGraphQuery` method here, this way, we would be able to compute tvl even if the graph becomes unreachable

**Review by g1nt0ki — CHANGES_REQUESTED** [review]
hi @alienHunterOnMars 

Thanks for the PR, it is mostly good, but can you split into three adapters, one for launchpad, amm and staking. This way, we could show the users tvl breakdown per product,  something like what we are doing for aftermath: https://defillama.com/protocol/aftermath-finance

**Comment by llamatester** [comment]
The adapter at projects/DegenHive-amm exports TVL: 
        
 
 
    sui                       600
    
    total                    600.091584263994

**Comment by llamatester** [comment]
The adapter at projects/DegenHive-dsui exports TVL: 
        
 
 
    sui                       242
    
    total                    241.5536144812

**Comment by llamatester** [comment]
The adapter at projects/DegenHive-memepad exports TVL: 
        
 
 
    sui                       77
    
    total                    77.4624824393

File: projects/DegenHive/index.js (hunk 1 above)
Comment (alienHunterOnMars) [reply projects/DegenHive/index.js]: done

**Comment by llamatester** [comment]
The adapter at projects/DegenHive-amm exports TVL: 
        
 
 
    sui                       599
    
    total                    598.7272819895079

**Comment by llamatester** [comment]
The adapter at projects/DegenHive-dsui exports TVL: 
        
 
 
    sui                       241
    
    total                    240.80344797659998

**Comment by llamatester** [comment]
The adapter at projects/DegenHive-memepad exports TVL: 
        
 
 
    sui                       77
    
    total                    77.22191572365

**Comment by g1nt0ki** [comment]
thanks for the update

**Pull request merged**